src/getMetroConfig.js

```diff
@@ -46,33 +46,48 @@ module.exports = symlinkedDependencies => {
 
     const extraNodeModules = peerDependenciesOfSymlinkedDependencies
         .map(mapModule)
-        .join(',\n')
+        .join(',\n  ')
 
-    const getBlacklistRE = symlinkedDependenciesPaths.map(mapPath).join(',\n')
+    const getBlacklistRE = symlinkedDependenciesPaths
+        .map(mapPath)
+        .join(',\n  ')
 
     const getProjectRoots = symlinkedDependenciesPaths
-        .map(path => `    path.resolve('${path}')`)
-        .join(',\n')
+        .map(path => `path.resolve('${path}')`)
+        .join(',\n  ')
 
     return dedent`
       const path = require('path');
-      const blacklist = require('metro/src/blacklist');
 
-      module.exports = {
-        extraNodeModules: {
-      ${extraNodeModules}
-        },
-        getBlacklistRE: () =>
-          blacklist([
-      ${getBlacklistRE}
-          ]),
-        getProjectRoots: () => [
-          // Include current package as project root
-          path.resolve(__dirname),
-          // Include symlinked packages as project roots
-      ${getProjectRoots}
-        ],
+      const extraNodeModules = {
+        ${extraNodeModules}
       };
-      
+      const blacklistRegexes = [
+        ${getBlacklistRE}
+      ];
+      const watchFolders = [
+        ${getProjectRoots}
+      ];
+
+      const metroVersion = require('metro/package.json').version;
+      const metroVersionComponents = metroVersion.match(/^(\\d+)\\.(\\d+)\\.(\\d+)/);
+      if (metroVersionComponents[1] === '0' && parseInt(metroVersionComponents[2], 10) >= 43) {
+          module.exports = {
+            resolver: {
+              extraNodeModules,
+              blacklistRE: require('metro-config/src/defaults/blacklist')(blacklistRegexes)
+            },
+            watchFolders
+          };
+      } else {
+          module.exports = {
+            extraNodeModules,
+            getBlacklistRE: () => require('metro/src/blacklist')(blacklistRegexes),
+            getProjectRoots: () => [path.resolve(__dirname)].concat(watchFolders)
+          };
+      }
```

Comment:
The crux of this change is down here. Before, the output format was:
```js
{
  extraNodeModules,
  getBlacklistRE,
  getProjectRoots
}
```

but now in RN 0.57, `getProjectsRoot` has been removed and replaced (as of RC3), `getBlacklistRE` has been replaced with just the RegEx, and the resolver config has moved to its own space:
```js
{
  resolver: {
    extraNodeModules,
    blacklistRE
  },
  watchFolders
}
```

The blacklist convenience function has moved too, so that's why I kept the require statement near the relevant config property.

Basically, this just detects whether we're on a newer version of Metro and returns the appropriate format.